pinot-core/src/test/java/com/linkedin/pinot/query/aggregation/groupby/AggregationGroupByTrimmingServiceTest.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.query.aggregation.groupby;
+
+import com.linkedin.pinot.common.response.broker.GroupByResult;
+import com.linkedin.pinot.core.query.aggregation.AggregationFunctionContext;
+import com.linkedin.pinot.core.query.aggregation.function.AggregationFunction;
+import com.linkedin.pinot.core.query.aggregation.function.AggregationFunctionFactory;
+import com.linkedin.pinot.core.query.aggregation.groupby.AggregationGroupByTrimmingService;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import org.apache.commons.lang.RandomStringUtils;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class AggregationGroupByTrimmingServiceTest {
+  private static final long RANDOM_SEED = System.currentTimeMillis();
+  private static final Random RANDOM = new Random(RANDOM_SEED);
+  private static final String ERROR_MESSAGE = "Random seed: " + RANDOM_SEED;
+
+  private static final AggregationFunction SUM = AggregationFunctionFactory.getAggregationFunction("SUM");
+  private static final AggregationFunctionContext[] AGGREGATION_FUNCTION_CONTEXTS =
+      {new AggregationFunctionContext(new String[]{"column"}, SUM)};
+  private static final AggregationFunction[] AGGREGATION_FUNCTIONS = {SUM};
+  private static final int GROUP_BY_TOP_N = 100;
+  private static final int NUM_GROUPS = 50000;
+
+  private List<String> _groups;
+  private AggregationGroupByTrimmingService _serverTrimmingService;
+  private AggregationGroupByTrimmingService _brokerTrimmingService;
+
+  @BeforeClass
+  public void setUp() {
+    Set<String> groupSet = new HashSet<>(NUM_GROUPS);
+    while (groupSet.size() < NUM_GROUPS) {
+      groupSet.add(
+          RandomStringUtils.randomAlphanumeric(RANDOM.nextInt(10)) + '\t' + RandomStringUtils.randomAlphanumeric(
```

Comment:
Could you also explicitly add groups with '\t'?
Also use random instead of randomAlphanumeric.